src/client/components/split-tile/split-tiles.tsx

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2017-2018 Allegro.pl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as React from "react";
+import { Colors } from "../../../common/models/colors/colors";
+import { Essence } from "../../../common/models/essence/essence";
+import { Split } from "../../../common/models/split/split";
+import { Stage } from "../../../common/models/stage/stage";
+import { Ternary, Unary } from "../../../common/utils/functional/functional";
+import { Fn } from "../../../common/utils/general/general";
+import { transformStyle } from "../../utils/dom/dom";
+import { SECTION_WIDTH } from "../../utils/pill-tile/pill-tile";
+import { TileOverflowContainer } from "../tile-overflow-container/tile-overflow-container";
+import { SplitTile } from "./split-tile";
+
+interface SplitTilesProps {
+  essence: Essence;
+  openedSplit: Split;
+  removeSplit: Unary<Split, void>;
+  updateSplit: Ternary<Split, Split, Colors, void>;
+  openMenu: Unary<Split, void>;
+  closeMenu: Fn;
+  dragStart: Ternary<string, Split, React.DragEvent<HTMLElement>, void>;
+  menuStage: Stage;
+  maxItems: number;
+  overflowOpen: boolean;
+  closeOverflowMenu: Fn;
+  openOverflowMenu: Fn;
+}
+
+export const SplitTiles: React.SFC<SplitTilesProps> = props => {
+  const { overflowOpen, closeOverflowMenu, openOverflowMenu, essence, maxItems, removeSplit, updateSplit, openedSplit, openMenu, closeMenu, dragStart, menuStage } = props;
+
+  const splits = essence.splits.splits.toArray();
+
+  const splitTiles = splits.map(split => {
+    const dimension = essence.dataCube.getDimension(split.reference);
+    return <SplitTile
+      key={split.toKey()}
+      split={split}
+      dimension={dimension}
+      removeSplit={removeSplit}
+      updateSplit={updateSplit}
+      open={split.equals(openedSplit)}
+      openMenu={openMenu}
+      closeMenu={closeMenu}
+      dragStart={dragStart}
+      containerStage={menuStage}
+      essence={essence} />;
```

Comment:
maybe `...props`?

Reply:
Not really, some props shouldn't be passed so you would need to cut them out, creating new object and still adding few more. I think that way it's more readable.